cmake/macros/macro_deal_ii_add_test.cmake

```diff
@@ -412,31 +412,46 @@ function(deal_ii_add_test _category _test_name _comparison_file)
 
       file(MAKE_DIRECTORY ${_test_directory})
 
+      #
+      # Determine whether the test shares a common executable target. This
+      # involves tests with .threads=N. and .mpirun=N. annotation, as well
+      # as tests with parameter files (that might share a common executable
+      # target).
+      #
+      # In this case we have to make sure that concurrently invoking the
+      # test does not accidentally trigger a concurrent build of the
+      # executable target. We ensure this by declaring an additional test
+      # that only builds the shared target / ensures the shared target is
+      # present. All run tests then requires this test target as a "setup
+      # fixture", see
+      # https://cmake.org/cmake/help/latest/prop_test/FIXTURES_REQUIRED.html#prop_test:FIXTURES_REQUIRED
+      #
+      set(_shared_target FALSE)
+      if(NOT "${_n_cpu}${_n_threads}" STREQUAL "00" OR "${_source_file}" MATCHES "(prm|json)$")
+        set(_shared_target TRUE)
+
+        #
+        # Build system-internal target name and final test name for the
+        # "executable" test. We have to make sure that the target and test
+        # names stay the same independent of test name and test category,
+        # thus the rather funny name:
+        #
+        set(_test_executable_target "test_dependency.${_target}.executable")
+        set(_test_executable_full   "test_dependency/${_target}.executable")
```

Comment:
That's hard to keep apart, with the difference only being whether you use a dot or a slash. Could we indicate the difference in a better way?

Reply:
This is directly adapted from what is already used as a naming convention in this file:
```
  380       set(_test_target    ${_category}.${_test_name}) # diff target name         
  381       set(_test_full      ${_category}/${_test_name}) # full test name           
  382       set(_test_directory ${CMAKE_CURRENT_BINARY_DIR}/${_test_name}.${_build_lowercase}) # directory to run the test in
```

I can replace `_full` by `_full_name` in the variable name if you want to. (`_test_full_test_name` sounds a bit silly.)

But in general I need a name for a top level target in the build system (that does not contain a slash `/`) and I need a name for the test in the form `category/test`. 

This is if we want to keep the `category/test` naming convention. I am happy to accomodate whatever but I suggest we make this part of a separate discussion and a separate pull request.

Reply:
Could we at least mark up the name of the executable as such then? Maybe `exe.${_category}.${_test_name}` or some such?